sconstruct.py

```diff
@@ -52,17 +52,18 @@ def shell(command):
           action="store_true",
           help="Compiles the code using Jumbo, this make it faster for clean builds")
 
-env = Environment(ENV=os.environ)
+env = Environment(ENV=os.environ, tools=["mingw"])
 env.Replace(CXX="g++")
-env.Append(CXXFLAGS="-Wall")
-env.Append(CXXFLAGS="-std=c++20")
-env.Append(CXXFLAGS="-Wno-unused -Wno-return-type-c-linkage")
-env.Append(CXXFLAGS=shell(["sdl2-config", "--cflags"]))
-env.Append(CXXFLAGS="-Iinc")
-env.Append(LINKFLAGS=shell(["sdl2-config", "--libs"]))
-env.Append(LINKFLAGS="-ldl")
-env.Append(LINKFLAGS="-rdynamic")  # Gives backtrace information
+env.MergeFlags("-Wall -Wno-unused -std=c++20")
+env.MergeFlags(shell(["sdl2-config", "--cflags"]))
+env.MergeFlags("-Iinc -Llib")
+env.MergeFlags(shell(["sdl2-config", "--libs"]))
+env.Append(LINKFLAGS=["-static-libgcc", "-static-libstdc++"])
+env.Append(LIBS="dl")
+env.Append(LIBS="gcc")
```

Comment:
Is `-lgcc` needed? I think we added it together with `-static-*` and didn't A/B test it.